src/main/java/jenkins/plugins/slack/webhook/WebhookHandler.java

```diff
@@ -0,0 +1,139 @@
+package jenkins.plugins.slack.webhook;
+
+
+import jenkins.model.Jenkins;
+
+import hudson.model.Build;
+import hudson.model.Result;
+import hudson.model.Project;
+import hudson.model.AbstractBuild;
+import hudson.model.AbstractProject;
+
+import hudson.security.ACL;
+
+import java.io.IOException;
+
+import java.util.List;
+import java.util.ArrayList;
+
+import org.kohsuke.stapler.interceptor.RequirePOST;
+
+import jenkins.plugins.slack.webhook.model.SlackPostData;
+import jenkins.plugins.slack.webhook.model.SlackTextMessage;
+import jenkins.plugins.slack.webhook.model.SlackWebhookCause;
+
+import org.acegisecurity.context.SecurityContext;
+import org.acegisecurity.context.SecurityContextHolder;
+
+
+
+public class WebhookHandler {
```

Comment:
I was actually thinking you might split this into separate command classes: ListProjectsCommand, GetProjectLogCommand and ScheduleJobCommand, implementing a simple interface with one "execute" method. Your router would then just lookup the command based on the "command text", call execute, and return whatever it gets back. This is fine too, but I thought I'd mention it in case you liked the idea.